sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py

```diff
@@ -95,7 +95,9 @@ def get_cached_access_token(self, scopes: Iterable[str], **kwargs: Any) -> Optio
             expires_on = int(token["expires_on"])
             if expires_on > int(time.time()):
                 refresh_on = int(token["refresh_on"]) if "refresh_on" in token else None
-                return AccessTokenInfo(token["secret"], expires_on, refresh_on=refresh_on)
+                return AccessTokenInfo(
+                    token["secret"], expires_on, token_type=token.get("token_type", "Bearer"), refresh_on=refresh_on
```

Comment:
What's the type of token? Is it possible a tuple?

Reply:
cache.search will return data in a dict.